### ISSUE TITLE
Issue with Exercise Accessibility on Mobile

### ISSUE THREAD

**Issue by Yuls2024** [freeCodeCamp/freeCodeCamp#57029]
### Describe the Issue

To whom it may concern,

I want to express my appreciation for freecodecamp.com—thank you for all the hard work you do.

I recently attempted to complete the exercises on my iPhone and encountered an issue. When I start Exercise 1 of Responsive Web Design or Step 1 of the tutorial, I cannot scroll down to reach the input line at the bottom where I'm supposed to enter my code. Unlike the desktop version, I'm unable to access this area.

Additionally, while there’s a button at the bottom of the screen, I cannot scroll down to click it, even when I rotate my phone sideways. As a result, it seems that completing the exercises on the mobile version is not currently feasible.

Thank you for looking into this matter.

Best regards,

Yulia

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-1

### Steps to Reproduce

1. Go to https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-1, on your mobile phone. I was testing it on iPhone 11 Pro Max and iPhone XR
2. Attempt to do that task.
3. Try to scroll down to the input line or the button.
4. You will not be able to do that.

### Expected behavior

I expected to scroll down and be able to find the input line for the exercise, just like in the desktop version. In desktop version it works.

I also expected to be able to scroll down to the button, but I'm not able to do that.

### Screenshots

![Image](https://github.com/user-attachments/assets/c1b11bca-7df6-4db9-a923-cf3f6a71265e)
![Image](https://github.com/user-attachments/assets/18cbdc86-82c4-4076-a681-10ceaa08dddf)
https://github.com/user-attachments/assets/abf32a0e-15ff-4f25-bb09-95f6d8fd9902

### System

- Device: iPhone 11 Pro Max and iPhone XR

### Additional context

I hope this helps.

**Comment by Feridun12** [comment]
Hello,

I would like to work on this issue. How can I contribute ? 

I do not see any open pull request option. Its my first time contributing to an open source project, I will be very glad if someone enlightens me.

**Comment by ilenia-magoni** [comment]
@Feridun12 this issue is not open for contribution as has not been triaged yet. The issues open for contribution have the "help wanted" label

**Comment by ojeytonwilliams** [comment]
Hi @Yuls2024, thanks for bringing this to our attention. 

Could you try scrolling by touching the left edge of the screen? What I suspect is happening is that the instructions are getting in the way and the scroll event is isn't making it to the editor underneath.

**Comment by ojeytonwilliams** [comment]
It's certainly not very friendly as is, so we do need to do one of those things.

Ideally have it scroll properly, since then we can use all the screen for useful things. One thing I noticed is that if the keyboard is visible, then swiping does what you'd expect, but if you dismiss it then swiping no longer scrolls. No idea why, as yet, but it's something to go on.

Also, there's one (not very nice) other option: make the scroll bar a little bigger. It exists, and will scroll the page, but I only noticed it because I saw it on desktop. Also, swiping to scroll is clearly the UX we should be aiming for.

**Comment by mmatsumoto1026** [comment]
> I'm wondering if we shouldn't make it a little wider?

I definately agree this idea, but the essence of this problem seems to be that campers have difficulty noticing the existence of the left side gutter, rather than whether it is easy for them to operate it.

Therefore, when a camper touches the editor, how about displaying a scroll progress indicator behind the gutter for a few seconds, and then escort them to scroll the editor by simply swiping the indicator.

In reality, campers are simply swiping on the gutter to scroll the editor, not swiping an indicator, which they can just see, but something like the timeline-scroller of the fCC forum post might help campers notice the gutter if the indicator will be displayed for a period of time.

> One thing I noticed is that if the keyboard is visible, then swiping does what you'd expect, but if you dismiss it then swiping no longer scrolls. No idea why, as yet, but it's something to go on.

It seems that the problem related to [the two vertical scroll bars issue](https://github.com/freeCodeCamp/freeCodeCamp/issues/48183) has reoccurred when the keyboard is displayed on a mobile device (such as a smartphone). In addition, currently [the tool panel buttons are not anchored to the bottom edge of viewport](https://github.com/freeCodeCamp/freeCodeCamp/pull/48586) when the keyboard is displayed on a mobile device as shown in below image. This is because the viewport height is not adjusted properly when the keyboard is visible on mobile devices, which causes the entire viewport to be off the screen.

These issues are in a different scope from the original issue, so It seems that we should open a new issue to discuss the solution for them.

I will also try to find a solution to these issue again.

<details>
<summary>the two vertical scroll bars issue reoccured and tool panel buttons are not anchored at the bottom of viewport</summary>

<img src=https://github.com/user-attachments/assets/04f4f0f4-b49a-4e88-8180-3860a275f5a6 width="50%" />
</details>

**Comment by ilenia-magoni** [comment]
An other issue with mobile accessibility seems to be scrolling right on snippets of code, see https://forum.freecodecamp.org/t/the-code-is-not-displayed-in-full/719094/

**Comment by Yuls2024** [comment]
> Hi [@Yuls2024](https://github.com/Yuls2024), thanks for bringing this to our attention.
> 
> Could you try scrolling by touching the left edge of the screen? What I suspect is happening is that the instructions are getting in the way and the scroll event is isn't making it to the editor underneath.

Hello @ojeytonwilliams, it did work when I tried to scroll on the left side, but not right away, I had to keep trying to scroll back and forth. And if you haven't mentioned it, I probably wouldn't have figured it out, because intuitively I usually always scroll on the right side.

**Comment by shootermv** [comment]
is it open for contribution now?

**Comment by ilenia-magoni** [comment]
@shootermv you should know that the `help wanted` label means that the issue is open for contribution

**Comment by shootermv** [comment]
If so, can I ask to assign myself to this issue?

**Comment by ilenia-magoni** [comment]
@shootermv issues are not usually assigned

**Comment by shootermv** [comment]
is it possible to reproduce this issue using mobile emultaor of chrome devtool ?

**Comment by a2937** [comment]
I have a semi-relevant question. Is there something preventing users from scrolling in the challenge prompt cause it certainly feels like there is?

**Comment by MrSaadMasood** [comment]
Not able to reproduce it on the chrome devtools.

https://github.com/user-attachments/assets/da894752-8f1d-4d6a-b3da-50816d21d29e

**Comment by gikf** [comment]
Reopening, since merged PR was creating UI issues #57626 and is reverted with #57630.

**Comment by akremdh1** [comment]
i reproduced the steps and the problem still exist
System
Device: iPhone 12 Pro Max 

https://github.com/user-attachments/assets/4476cd13-0799-4d72-906e-781e150cfd39

**Comment by ojeytonwilliams** [comment]
Hey, we're focusing on adding all of the curriculum into the mobile app which has a much better UX.
